data/json/effect_on_condition.json

```diff
@@ -302,5 +302,214 @@
+  {
+    "type": "effect_on_condition",
+    "id": "portal_strengthen",
+    "recurrence_min": "30 minutes",
+    "recurrence_max": "1 hours",
+    "condition": { "is_weather": "portal_storm" },
+    "deactivate_condition": { "not": { "is_weather": "portal_storm" } },
+    "effect": [
+      { "u_message": "Debug: Portals get stronger.", "type": "bad" },
+      { "u_adjust_var": "portal_storm_strength", "type": "weather", "context": "weather_effect", "adjustment": 1 },
+      { "u_lose_var": "portal_storm_active", "type": "weather", "context": "cause" }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "portal_message",
+    "recurrence_min": 15,
+    "recurrence_max": 60,
+    "condition": { "and": [ { "one_in_chance": 10 }, { "is_weather": "portal_storm" }, "u_is_outside", "u_can_see" ] },
+    "deactivate_condition": { "not": { "is_weather": "portal_storm" } },
+    "effect": [ { "u_message": "PORTAL_STORM", "snippet": true, "type": "bad" } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "radiation_portal",
+    "recurrence_min": 30,
+    "recurrence_max": 60,
+    "condition": {
+      "and": [
+        { "one_in_chance": 10 },
+        { "is_weather": "portal_storm" },
+        "u_is_outside",
+        { "not": { "u_has_worn_with_flag": "RAD_PROOF" } },
+        {
+          "u_compare_var": "portal_storm_strength",
+          "type": "weather",
+          "context": "weather_effect",
+          "op": ">=",
+          "value": 3
+        }
+      ]
+    },
+    "deactivate_condition": { "not": { "is_weather": "portal_storm" } },
+    "effect": [
+      { "u_message": "Debug:A portal opens for an instant and a painful light pours out.", "type": "bad" },
+      { "u_mod_radiation": 10 }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "monster_portal_wraith_test",
+    "recurrence_min": 30,
+    "recurrence_max": 60,
+    "effect": [
+      { "u_message": "A portal opens for an instant and closes after a monster exits.", "type": "bad" },
+      { "u_set_spawn_monster": "mon_wraith", "real_count": 1, "hallucination_count": 1, "lifespan_min":"5 minutes","lifespan_max":"10 minutes" }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "monster_portal_wraith",
+    "recurrence_min": 30,
+    "recurrence_max": 60,
+    "condition": {
+      "and": [
+        { "one_in_chance": 10 },
+        { "is_weather": "portal_storm" },
+        "u_is_outside",
+        {
+          "u_compare_var": "portal_storm_strength",
+          "type": "weather",
+          "context": "weather_effect",
+          "op": ">=",
+          "value": 1
+        }
+      ]
+    },
+    "deactivate_condition": { "not": { "is_weather": "portal_storm" } },
+    "effect": [
+      { "u_message": "A portal opens for an instant and closes after a monster exits.", "type": "bad" },
+      { "u_set_spawn_monster": "mon_wraith", "real_count": 1, "hallucination_count": 1, "lifespan_min":"5 minutes","lifespan_max":"10 minutes" }
```

Comment:
So 1 real, 1 hallucination?

Reply:
For now.  At some point I want those numbers to change with storm strength but I might wait for #50305 so I can have it be some fraction of storm strength.